### PR TITLE
limit the delay before next try in S3

### PR DESCRIPTION
**Pull request by CheSema**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

It is make no sense to sleep hours between retries. Let's limit the delay. When delay is limited `s3_max_redirects` could be set to reasonable high value.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit d9e15c00c99c45bdba546173bbcfaf1698196404 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/54651/d9e15c00c99c45bdba546173bbcfaf1698196404/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

**Comment by CheSema** [comment]
Stateless tests (release, s3 storage):
 02443_detach_attach_partition  -- https://github.com/ClickHouse/ClickHouse/issues/54748

File: src/IO/S3/Client.cpp [hunk 1]
```diff
@@ -846,7 +837,8 @@ std::unique_ptr<S3::Client> ClientFactory::create( // NOLINT
             std::move(credentials),
             credentials_configuration);
 
-    client_configuration.retryStrategy = std::make_shared<Client::RetryStrategy>(std::move(client_configuration.retryStrategy));
+    client_configuration.retryStrategy = std::make_shared<Client::RetryStrategy>(client_configuration.s3_retry_attempts);
```
Comment (kssenii) [inline src/IO/S3/Client.cpp:840]: I think `MaxDelayMs` should also be changeable by setting here.

**Review by kssenii — APPROVED** [review]
lgtm, but one small comment

File: src/IO/S3/Client.cpp (hunk 1 above)
Comment (CheSema) [reply src/IO/S3/Client.cpp:840]: I will add a dedicated setting for it when I find a reason to customize it from node to node.

**Pull request merged**